### DISCUSSION TITLE
Parsing query strings?

### DISCUSSION THREAD

**Discussion by msr8** [pyscript/pyscript#1525] [Q&A] [answered]
Is it possible to parse query strings in pyscript? I suppose I could use `js.window.location.search`, but is there any "pythonic" way to do so?

Edit: The urllib.parse python module also help achieve this ([Demonstration](https://stackoverflow.com/a/46601122/17002774))

**Comment by WebReflection** [comment, marked as answer]
as you eventually need to reach for `location.search` anyway, I suggest you also use the right primitive to deal with query string parameters, which is [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)

```python
from js import location, URLSearchParams

qs = URLSearchParams.new(location.search)
```

That being said, it's a nice idea to expose somehow a utility that simplifies this dance and results into a python dictionary of query string parameters, but that could also be a handy module, not sure it's planned (or actually in core) already.

**Reply by msr8** [reply]
Thank you for the answer! Did some digging and found that urllib.parse module contains `parse_qs()` and `parse_qsl()` ([Demonstration](https://stackoverflow.com/a/46601122/17002774)) that help do this process and `dict( urllib.parse.parse_qsl(js.location.search[1:]) )` returns a python dictionary of the query string keys and values

**Reply by WebReflection** [reply]
very nice and yes, I think that's the best Pythonic way of doing it, a lot less indirection too!
